testsuite/forge/src/backend/local/swarm.rs

```diff
@@ -541,3 +546,41 @@ impl Swarm for LocalSwarm {
         todo!()
     }
 }
+
+#[derive(Debug)]
+pub struct ActiveNodesGuard {
+    counter: Arc<Mutex<usize>>,
+    slots: usize,
+}
+
+impl ActiveNodesGuard {
+    pub async fn grab(slots: usize, counter: Arc<Mutex<usize>>) -> Self {
+        let max = num_cpus::get();
+        loop {
+            {
+                let mut guard = counter.lock();
+                if *guard < max {
+                    info!(
+                        "Grabbed {} node slots to start test, already active {} swarm nodes",
+                        slots, *guard
+                    );
+                    *guard += slots;
+                    drop(guard);
+                    return Self { counter, slots };
+                }
+                info!(
+                    "Too many active swarm nodes ({}), waiting to start {} new ones",
+                    *guard, slots,
+                );
+            }
+            tokio::time::sleep(Duration::from_secs(2)).await;
```

Comment:
Instead of busy loop here, use of a channel will be ideal for performance reason but I guess this is just test, so this  should be okay.